### PR TITLE
MSQ: Fix task lock checking during publish, fix lock priority.

### PR DESCRIPTION
**Pull request by gianm**
Fixes two issues:

1) ControllerImpl did not properly check the return value of
   SegmentTransactionalInsertAction when doing a REPLACE. This could cause
   it to not realize that its locks were preempted.

2) Task lock priority was the default of 0. It should be the higher
   batch default of 50. The low priority made it possible for MSQ tasks
   to be preempted by compaction tasks, which is not desired.

### CODE REVIEW

**Review by kfaraz — APPROVED** [review]
LGTM

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java [hunk 1]
```diff
@@ -204,6 +204,12 @@ public void stopGracefully(final TaskConfig taskConfig)
     }
   }
 
+  @Override
+  public int getPriority()
+  {
+    return getContextValue(Tasks.PRIORITY_KEY, Tasks.DEFAULT_BATCH_INDEX_TASK_PRIORITY);
```
Comment (cryptoe) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java:210]: Nit: should we document this here https://druid.apache.org/docs/24.0.0/ingestion/tasks.html#lock-priority ?

**Review by cryptoe — APPROVED** [review]
Minor nit: LGTM otherwise.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java (hunk 1 above)
Comment (gianm) [reply extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java:210]: 👍  updated.

**Comment by cryptoe** [comment]
LGTM thanks @gianm 🚀

**Pull request merged**